src/common/pa_converters.c

```diff
@@ -1073,16 +1072,19 @@ static void Int32_To_UInt8_Dither(
     void *sourceBuffer, signed int sourceStride,
     unsigned int count, struct PaUtilTriangularDitherGenerator *ditherGenerator )
 {
-    /* PaInt32 *src = (PaInt32*)sourceBuffer;
-    unsigned char *dest =  (unsigned char*)destinationBuffer; */
-    (void)ditherGenerator; /* unused parameter */
+    PaInt32 *src = (PaInt32*)sourceBuffer;
+    unsigned char *dest =  (unsigned char*)destinationBuffer;
+    PaInt32 dither;
 
     while( count-- )
     {
-        /* IMPLEMENT ME */
+        /* REVIEW */
+        dither = PaUtil_Generate16BitTriangularDither( ditherGenerator );
 
-        /* src += sourceStride;
-        dest += destinationStride; */
+        *dest = (unsigned char)((((*src) >> 1) + dither) >> 23);
```

Comment:
UInt8 audio is centered around 128 and ranges from 0 to 255.
So silence would be all 128s.

Reply:
To clarify, you need to add an offset to convert Int32 to Uint8 format.